src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -387,9 +385,7 @@ public static function get_block_data() {
 		 * @param WP_Theme_JSON_Data $theme_json Class to access and update the underlying data.
 		 */
 		$theme_json = apply_filters( 'wp_theme_json_data_blocks', new WP_Theme_JSON_Data( $config, 'blocks' ) );
-		$config     = $theme_json->get_data();
-
-		static::$blocks = new WP_Theme_JSON( $config, 'blocks' );
```

Comment:
Ditto here and also below about the different `$origin` here (`blocks` instead of `theme`)